### PR TITLE
Change to Pandas implementation to keep the order

### PR DESCRIPTION
**Pull request by ReinierKoops**
Merging this will preserve the order of the features as suggested by @PaulZhutovsky .

### CODE REVIEW

**Comment by ReinierKoops** [comment]
#196 , this is an improvement on the previous implementation improvement.

File: tests/feature_elimination/test_feature_elimination.py [hunk 1]
```diff
@@ -337,6 +339,75 @@ def test_get_feature_shap_values_per_fold(X, y):
...
+    Test a usecase which appears to be flickering with probatus 1.8.6 and lower.
+
+    Expected result: every run a different outcome.
+    """
+    SEED = 1234
+
+    feature_names = [
+        "f1",
+        "f2",
+        "f3_static",
```
Comment (gverbock) [inline tests/feature_elimination/test_feature_elimination.py:353]: why "static" and in "f3_static"? 

What about

```
feature_names = [f"f{num}" for num in range(1, 21}]
```

with eventually a rename if you need "f3_static"

File: probatus/feature_elimination/feature_elimination.py [hunk 2]
```diff
@@ -526,11 +526,12 @@ def fit(
             # Get current dataset info
             current_features_set = remaining_features
             if columns_to_keep is None:
-                remaining_removeable_features = sorted(list(set(current_features_set)))
+                remaining_removeable_features = pd.Series(current_features_set).unique()
             else:
-                remaining_removeable_features = sorted(
-                    list(set(current_features_set) | set(columns_to_keep))
-                )
+                remaining_removeable_features = pd.Series(
```
Comment (gverbock) [inline probatus/feature_elimination/feature_elimination.py:531]: Would it be nice to comment that this is for freezing the order?

File: tests/feature_elimination/test_feature_elimination.py [hunk 3]
```diff
@@ -337,6 +339,75 @@ def test_get_feature_shap_values_per_fold(X, y):
     assert train_score > 0.9
     assert shap_values.shape == (2, 3)
 
+def test_shap_rfe_same_features_are_kept_after_each_run():
+    """
+    Test a usecase which appears to be flickering with probatus 1.8.6 and lower.
```
Comment (gverbock) [inline tests/feature_elimination/test_feature_elimination.py:344]: usecase => use case

**Review by gverbock — COMMENTED** [review]
@ReinierKoops  I made some small comments on the PR

File: tests/feature_elimination/test_feature_elimination.py (hunk 1 above)
Comment (ReinierKoops) [reply tests/feature_elimination/test_feature_elimination.py]: Fixed.

File: probatus/feature_elimination/feature_elimination.py (hunk 2 above)
Comment (ReinierKoops) [reply probatus/feature_elimination/feature_elimination.py:533]: Fixed.

File: tests/feature_elimination/test_feature_elimination.py (hunk 3 above)
Comment (ReinierKoops) [reply tests/feature_elimination/test_feature_elimination.py]: Fixed.

**Pull request merged**